lib/spi/types.ts

```diff
@@ -82,6 +82,12 @@ export interface ApplicationRepository extends GitRepositoryReference {
     credentialsType?: "USERNAME" | "TOKEN" | "SSH"
 
 }
+/**
+ * Adds Constraints to application repository
+ */
+export class ApplicationRepositoryConstraints implements constraints.ConstraintsType<ApplicationRepository> {
+    credentialsSecretName = new constraints.InternetHostStringConstraint();
```

Comment:
Shouldn't it be allowed to use secrets named as `/${product}/${stage}/cluster-bootstrap/${repo-credentials}` ?

Reply:
It will require a mapping logic to transform that name to the k8s secret name. The current design is that the secret you create in AWS Secrets Manager is propagated as is (with the same name) to EKS. The k8s secret name constraint is:

> You can create a name with a maximum length of 253 characters using only alphanumeric characters and hyphens. Names cannot start with a hyphen and the alpha characters can only be lowercase.

Please feel free to create a feature request to implement this logic for the name translation.